crates/solver/src/solver/score_computation.rs

```diff
@@ -0,0 +1,135 @@
+// Suppose a solver generates a solution, and we compute the three variables:
+// surplus, fees and gas_cost. Currently, the objective is then simply equal to
+// surplus + fees - gas_cost. However, for CIP-20, we now have one additional
+// step for each solver. So for each solver we have trained a revert risk model,
+// which essentially is a function that computes the probability of the
+// settlement not reverting, and is a function of 3 parameters.
+// p (num_orders_in_sol, gas_used, gas_price).
+// So we compute this p, and then the score is equal to:
+// score = p * (surplus + fees) - gas_cost
+
+use {
+    super::SolverType,
+    crate::objective_value,
+    anyhow::{Context, Result},
+    clap::{Parser, ValueEnum},
+    ethcontract::U256,
+    num::ToPrimitive,
+    std::{
+        collections::HashMap,
+        fmt::{Display, Formatter},
+        ops::Neg,
+        str::FromStr,
+    },
+};
+
+#[derive(Debug, Default, Clone)]
+pub struct ScoreCalculator {
+    a: f64,
+    b: f64,
+    c: f64,
+    x: f64,
```

Comment:
I think either these need nicer names (e.g. `a -> gas_amount_factor`) or some comment which describes the revert probability function which relates the score computation inputs to these parameters.

Reply:
renamed. @harisang what should I call the parameter `x`?

Reply:
So, in the logistic regression literature it is called `intercept`. Not sure if that adds helps, but I guess it is more descriptive compared to `x`

Reply:
Renamed